src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,196 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size = 0;
+    private Node<T> head;
+    private Node<T> tail;
+
+    private static class Node<T> {
+        private T value;
+        private Node<T> prev;
+        private Node<T> next;
+
+        public Node(Node<T> prev, T value, Node<T> next) {
+            this.value = value;
+            this.prev = prev;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        size++;
+        final Node<T> l = tail;
+        final Node<T> newNode = new Node<>(l, value, null);
+        tail = newNode;
+        if (l == null) {
+            head = newNode;
+        } else {
+            l.next = newNode;
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        size++;
+        checkPositionIndex(index);
+
+        Node<T> newNode = defaultNode(value);
+
+        if (index == 0) {
+            newNode.next = head;
+            head = newNode;
+        } else if (index == size - 1) {
+            node(size - 2).next = newNode;
+            newNode.prev = node(size - 2);
+            tail = newNode;
+        } else {
+            Node<T> prevNode = node(index - 1);
+            newNode.next = prevNode.next;
+            newNode.prev = prevNode;
+            prevNode.next.prev = newNode;
+            prevNode.next = newNode;
+        }
+
     }
 
     @Override
     public void addAll(List<T> list) {
```

Comment:
just iterate over that list and call `add(T value)` method instead of this logic